### PR TITLE
Restore Puppet 3.4 and earlier compatibility

### PR DESCRIPTION
**Pull request by mhaskel**

### CODE REVIEW

File: metadata.json [hunk 1]
```diff
@@ -27,11 +27,11 @@
   "requirements": [
     {
       "name": "pe",
-      "version_requirement": "3.x"
+      "version_requirement": ">= 3.3.0"
```
Comment (daenney) [inline metadata.json]: Why?

**Comment by daenney** [comment]
This looks pretty awesome @mhaskel <3.

The only thing I worry a bit about now is that none of the tests cover the pre-3.5.0 behaviour. But that's going to be hard to do with unit tests and the strict vars mess since we're only faking the Puppet version.

File: metadata.json (hunk 1 above)
Comment (mhaskel) [inline metadata.json]: stdlib couldn't be upgraded to 4.x until PE 3.3

Comment (daenney) [inline metadata.json]: I see. Oh well.

**Comment by mhaskel** [comment]
@daenney I did add one unit test to just make sure params didn't completely explode with puppetversion = 3.4.something.

I ran acceptance tests manually with trusty and 3.4.3, and it passed now (and failed without the patch)

**Comment by daenney** [comment]
Good enough for me. You have my :+1:.

**Pull request merged**